### PR TITLE
File explorer add support for force deleting a non-empty directory

### PR DESCRIPTION
**Pull request by AdityaHegde**
- [x] Add a confirmation to force delete a non-empty directory
- [x] Deleting a folder for an active file should redirect to `/`

### CODE REVIEW

**Review by begelundmuller — APPROVED** [review]
Approving from backend perspective

File: web-common/src/features/file-explorer/ForceDeleteConfirmation.svelte [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+  function handleClose() {
+    open = false;
+  }
+</script>
+
+<Dialog on:close={handleClose} {open}>
+  <div slot="title">
+    Folder is not empty. Are you sure you want to delete everything?
+  </div>
```
Comment (ericpgreen2) [inline web-common/src/features/file-explorer/ForceDeleteConfirmation.svelte:16]: Can we change the title to: "Are you sure you want to delete everything?"

And add a `body` section: "This folder is not empty. All contained items will be deleted."

File: web-common/src/features/file-explorer/ForceDeleteConfirmation.svelte [hunk 2]
```diff
@@ -0,0 +1,28 @@
+<script lang="ts">
+  import { Button } from "@rilldata/web-common/components/button";
+  import Dialog from "@rilldata/web-common/components/dialog/Dialog.svelte";
```
Comment (ericpgreen2) [inline web-common/src/features/file-explorer/ForceDeleteConfirmation.svelte:3]: (Nice-to-have, but by no means a blocker): it looks like ShadCN-Svelte has a specific [Alert Dialog component](https://www.shadcn-svelte.com/docs/components/alert-dialog) for these use cases.

File: web-common/src/features/file-explorer/ForceDeleteConfirmation.svelte
Comment (ericpgreen2) [inline web-common/src/features/file-explorer/ForceDeleteConfirmation.svelte:1]: Nit: mind including `Dialog` in the component name, e.g. `ForceDeleteConfirmationDialog.svelte`?

File: web-common/src/features/file-explorer/transform-file-list.ts [hunk 3]
```diff
@@ -61,3 +62,12 @@ export function transformFileList(files: V1DirEntry[]): Directory {
 
   return rootDirectory;
 }
+
+export function findDirectory(root: Directory | undefined, filePath: string) {
```
Comment (ericpgreen2) [inline web-common/src/features/file-explorer/transform-file-list.ts:66]: I think it'd be simpler if this didn't accept an `undefined` root directory

File: web-common/src/features/file-explorer/utils.ts [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+export function isCurrentActivePage(
+  filePath: string,
+  currentFile: string | undefined,
+  isDir: boolean,
+) {
+  filePath = removeLeadingSlash(filePath);
+  currentFile = removeLeadingSlash(currentFile ?? "");
+  return (
+    currentFile !== "" &&
```
Comment (ericpgreen2) [inline web-common/src/features/file-explorer/utils.ts:11]: Adding a guard clause at the top of the function could be better:
```
if (currentFile === undefined) return false
```

Further, I'd wonder if the function even needs to accept an `undefined` `currentFile`.

**Review by ericpgreen2 — CHANGES_REQUESTED** [review]
I played around with it, and UXQA-wise:
1. I'm not able to delete a non-empty directory. JAM: https://jam.dev/c/089c392e-78d9-4144-9470-a3a93be0173f
2. I'm seeing this error in the console
image

File: web-common/src/features/file-explorer/ForceDeleteConfirmation.svelte (hunk 2 above)
Comment (AdityaHegde) [reply web-common/src/features/file-explorer/ForceDeleteConfirmation.svelte:3]: Good idea. Seems like it has some dependancy on button. Lets add it in a separate PR and move all alerts to use this.

File: web-common/src/features/file-explorer/utils.ts (hunk 4 above)
Comment (AdityaHegde) [reply web-common/src/features/file-explorer/utils.ts:11]: `$page.params.file` could be undefined when user is on home. Moving that check here for now. Felt cleaner to deal with `$page` here itself.

**Comment by AdityaHegde** [comment]
I couldnt reproduce the 1st problem. Did you restart the backend server after checking this code out?

**Comment by ericpgreen2** [comment]
One UX issue:

When viewing a file and force deleting its directory, I see our current 404/error page. I'd expect to be re-directed to root. JAM: https://jam.dev/c/0d3b3c42-88c0-45be-878e-a209819d38d6

**Review by ericpgreen2 — APPROVED** [review]

**Pull request merged**